internal/gw/client.go

```diff
@@ -60,7 +60,28 @@ func (c *client) doJSONRequest(ctx context.Context, method string, url string, b
 	defer resp.Body.Close()
 
 	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		// TODO: attempt to read body here to log something about the failure
+		var b struct {
```

Comment:
When filing the issue I didn't expect we'd be parsing the response as JSON.  Can we please just log the bytes instead?

There are a few issues with parsing it as JSON:

- I don't feel confident that we have the error format under control throughout all of exodus-gw.  For example, our docs at https://release-engineering.github.io/exodus-gw/api.html#operation/update_publish_items__env__publish__publish_id__put say that "detail" is going to be an array there, not a string.  If we're going to start parsing it as string then I think we'd better file an issue to make the error format consistent everywhere (or just fix the docs if they're wrong).
- The error is not necessarily coming from exodus-gw.  Keep in mind that exodus-gw is behind platform-sidecar which is behind the openshift router and each of those layers can have their own reasons to fail.  So there are several layers besides exodus-gw which can return an error response, and *even if* we're 100% confident in the error format from exodus-gw, we still have to be prepared to get responses of unknown error formats... and if that happens, we'd want to see the error rather than throw it away.
- More code paths in need of testing (I notice the PR currently didn't meet 100% coverage)

Due to above issues I'd say that, at minimum, if we can't parse as JSON then we'd better just log the bytes.  Personally I would make it *only* log the bytes for now (we can always come back and add friendlier parsing of specific errors later if it seems worthwhile).

Reply:
Sounds good!